src/components/cdTab.js

```diff
@@ -1,34 +1,183 @@
+    setLoading(isLoading)
+    setIsError(isError)
+  }, [
+    componentDescriptor,
+    isLoading,
+    isError,
+    setRaw,
+    setLoading,
+    setIsError,
+  ])
+}
+FetchRawComponent.displayName = 'FetchRawComponent'
+FetchRawComponent.propTypes = {
+  setRaw: PropTypes.func.isRequired,
+  setLoading: PropTypes.func.isRequired,
+  setIsError: PropTypes.func.isRequired,
+  componentName: PropTypes.string.isRequired,
+  componentVersion: PropTypes.string.isRequired,
+  ocmRepo: PropTypes.string,
+  versionFilter: PropTypes.string,
+}
+
+
+const Options = ({
+  isLoading,
+  raw,
+  setRaw,
+}) => {
+  return <Box>
+    <FormGroup>
+      <FormControlLabel
+        control={
+          <Tooltip
+            title='Raw document as retrieved from OCM repository, without (de)serialisation'
+          >
+            <Switch
+              checked={raw}
+              // eslint-disable-next-line no-unused-vars
+              onChange={(event) => setRaw((prev) => !prev)}
+              disabled={isLoading}
+            />
+          </Tooltip>
+        }
+        label='show raw'
+      />
+    </FormGroup>
+  </Box>
+}
+Options.displayName = 'Options'
+Options.propTypes = {
+  isLoading: PropTypes.bool,
+  raw: PropTypes.bool.isRequired,
+  setRaw: PropTypes.func.isRequired,
+}
+
+
+const Viewer = ({
+  componentDescriptor,
+  isLoading,
+  isError
+}) => {
+  if (isError) return <Alert severity='critical'>
+    Something went wrong
+  </Alert>
+
+  if (isLoading || !componentDescriptor) return <Box
+    display='flex'
+    justifyContent='center'
+  >
+    <CenteredSpinner/>
+  </Box>
+
+  return <ObjectTextViewer
+    obj={componentDescriptor}
+  />
+}
+Viewer.displayName = 'Viewer'
+Viewer.propTypes = {
+  isLoading: PropTypes.bool,
+  componentDescriptor: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
+  isError: PropTypes.bool,
+}
 
 
 export const CdTab = ({
   componentDescriptor,
   isLoading,
+  ocmRepo,
+  versionFilter,
 }) => {
-  if (!isLoading) return <ObjectTextViewer obj={componentDescriptor}/>
+  const [rawMode, setRawMode] = React.useState(false)
+
+  const [raw, setRaw] = React.useState()
+  const [rawLoading, setRawLoading] = React.useState()
+  const [rawIsError, setRawIsError] = React.useState()
+
+  const _isLoading = () => {
+    if (rawMode) return rawLoading
+    else return isLoading
+  }
+
+  const _isError = () => {
+    if (rawMode) return rawIsError
+    else return false // handled earlier already
+  }
+
+  const _componentDescriptor = () => {
+    if (rawMode) return raw
+    else return componentDescriptor
+  }
 
-  return <Grid container>
+  return <Box
+    display='flex'
+    flexDirection='column'
+  >
     {
-      Array.from(Array(30).keys()).map(e => {
-        return <Grid
-          item
-          xs={6}
-          key={e}
-        >
-          <Skeleton/>
-        </Grid>
-      })
+      // use bool to prevent render of value "0"
+      Boolean(rawMode & !raw) && <FetchRawComponent
```

Comment:
I suppose you meant to use `&&` instead? 🤔

Reply:
No.
This condition prevents to fetch raw component-descriptor again, if it is already present.